src/Compilers/CSharp/Test/Symbol/Symbols/SymbolErrorTests.cs

```diff
@@ -19412,6 +19412,142 @@ public void AbstractInSubmission()
+        public const string Value = ""TEST VALUE"";
+    }
+}";
+            var forwardedToReference = CreateCompilationWithMscorlib(forwardedToCode, assemblyName: "Destination").EmitToImageReference();
+
+            var forwardingCode = @"
+namespace ForwardingNamespace
+{
+    public class Program
+    {
+        public static void Main()
+        {
+            System.Console.WriteLine(Destination.TestClass.Value);
+        }
+    }
+}";
+            var forwardingIL = @"
+.assembly extern Destination { }
+.class extern forwarder Destination.TestClass
+{
+	.assembly extern Destination
+}
+.class extern forwarder Destination.TestClass
+{
+	.assembly extern Destination
+}";
+            var forwardingCompilation = CreateCompilationWithCustomILSource(
+                source: forwardingCode,
+                ilSource: forwardingIL,
+                references: new MetadataReference[] { forwardedToReference },
+                options: new CSharpCompilationOptions(OutputKind.ConsoleApplication));
+
+            CompileAndVerify(forwardingCompilation, expectedOutput: "TEST VALUE");
+        }
+
+        [Fact, WorkItem(16484, "https://github.com/dotnet/roslyn/issues/16484")]
+        public void MultipleForwardsToDifferentAssembliesWithoutReferencingThemShouldIgnoreIt()
+        {
+            var forwardingIL = @"
+.assembly extern Destination1 { }
+.assembly extern Destination2 { }
+.class extern forwarder Destination.TestClass
+{
+	.assembly extern Destination1
+}
+.class extern forwarder Destination.TestClass
+{
+	.assembly extern Destination2
+}";
+            var forwardingCompilation = CreateCompilationWithCustomILSource(string.Empty, forwardingIL);
+
+            forwardingCompilation.VerifyDiagnostics();
+        }
+
+        [Fact, WorkItem(16484, "https://github.com/dotnet/roslyn/issues/16484")]
+        public void MultipleForwardsToDifferentAssembliesWhileReferencingThemShouldErrorOut()
+        {
+            var forwardingCode = @"
+namespace ForwardingNamespace
+{
+    public class Program
+    {
+        public static void Main()
+        {
+            new Destination.TestClass();
+        }
+    }
+}";
+            var forwardingIL = @"
+.assembly extern Destination1 { }
+.assembly extern Destination2 { }
+.class extern forwarder Destination.TestClass
+{
+	.assembly extern Destination1
+}
+.class extern forwarder Destination.TestClass
+{
+	.assembly extern Destination2
+}";
+            var forwardingCompilation = CreateCompilationWithCustomILSource(forwardingCode, forwardingIL);
+
+            forwardingCompilation.VerifyDiagnostics(
+                // (8,29): error CS1577: Type 'Destination.TestClass' is forwarded to multiple assemblies: 'Destination1' and 'Destination2'
+                //             new Destination.TestClass();
+                Diagnostic(ErrorCode.ERR_TypeForwardedToMultipleAssemblies, "TestClass").WithArguments("Destination.TestClass", "Destination1", "Destination2").WithLocation(8, 29));
+        }
```

Comment:
Maybe add a test with more than 2 duplicate type forwards (in that case we still expect only one pair of duplicates to be reported).